include/test/mir/test/doubles/multi_logger.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2022 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_TEST_DOUBLES_MULTI_LOGGER_H_
+#define MIR_TEST_DOUBLES_MULTI_LOGGER_H_
+
+#include "mir/logging/logger.h"
+
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace mir
+{
+namespace test
+{
+namespace doubles
+{
+class MultiLogger : public mir::logging::Logger
+{
+public:
+    void add(std::shared_ptr<mir::logging::Logger> logger);
```

Comment:
A mutable `Logger` leaves a lot of potential for threading issues.

How about:
```
    MultiLogger(std::initializer_list<std::shared_ptr<logging::Logger>> loggers);
```

(And, maybe, `unique_ptr` is better)